### ISSUE TITLE
Tagging & Filtering feature

### ISSUE THREAD

**Issue by wada3n** [phase1geo/Outliner#44]
- A tag starts with **@** character or **$**.
 Tagging helps you to organize and categorize your outline, separate important and unimportant information. For example: `@important, @todo, @today.`
  - To filter your outline based on a tag, just click on that tag. and tha app will fold the content and display all the lines containing the tag along with the headline structure above it.

This idea is present on Outlinely Mac 

![May-19-2020 19-20-19](https://user-images.githubusercontent.com/13304792/82363795-3e10ef80-9a06-11ea-99b6-d12a3ce15d47.gif)

**Comment by phase1geo** [comment]
Looks pretty cool and useful.

**Comment by phase1geo** [comment]
I have an implementation of this feature available in the development branch if you want to give it a go and let me know what you think.  There may be a few tweaks to the display of the filtered nodes before the next stable release, but I think I'm reasonably happy with the interface.

**Comment by wada3n** [comment]
@phase1geo  You're on fire today, wow, this is special :) 
I think this is a very good start, when adding the filtering by clicking on the tag will be a beautiful thing.

There is a simple problem: when selecting a tag and then select Clear all formatting. The red color of the tag is Disappears!

**Comment by phase1geo** [comment]
I will get the clear formatting issue fixed ASAP.

You can filter rows by tag by Control-clicking on the tag.

**Comment by phase1geo** [comment]
Should be fixed now.

**Comment by wada3n** [comment]
I can confirm that.
Everything works great
You deserve all gratitude

**Comment by wada3n** [comment]
By the way, when pressing the tag, it would be better if the root row was added for each child, the problem is that if the lines are the same, things will mix completely:

e.g:
- Collapse and expand
      - Introduction `@test`
- Notes
      - Introduction  `@test`
- Completed Status
      - Introduction  `@test`

When you press Tag, you can visualize it

**Comment by jaslar** [comment]
Cool idea. I'm finding that when I control-click on a tag, I don't always get the first one in the list. That is, I have a list of four rows with the tag. 

First, I created three years, with the same @new tag. If I control-clicked on the tag, I get all three.
Second, I added a new row above the previous three. When I control-click on the first one, I see only the first one. I click on the last one, I get the last three, but not the first one?
